rosidl_runtime_c/test/benchmark_string_conversion.cpp

```diff
@@ -0,0 +1,73 @@
+// Copyright 2020 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <string>
+
+#include "rosidl_runtime_c/string_functions.h"
+#include "rosidl_runtime_c/u16string_functions.h"
+
+#include "performance_test_fixture/performance_test_fixture.hpp"
+
+using performance_test_fixture::PerformanceTest;
+
+BENCHMARK_DEFINE_F(PerformanceTest, string_assign_complexity)(benchmark::State & st)
+{
+  size_t len = st.range(0);
+  std::string data(len, '*');
+
+  rosidl_runtime_c__String s;
+  if (!rosidl_runtime_c__String__init(&s)) {
+    st.SkipWithError("String initialization failed");
+    return;
+  }
+
+  // No resize function - just do a copy
+  rosidl_runtime_c__String__assignn(&s, data.c_str(), len);
```

Comment:
Since this one doesn't need an explicit resize, it might be descriptive to have another benchmark for assigning to a string of a different size. I.e., line 36 assigns a short string (or empty one), then 39 assigns the string of `len`. Or start with a long string and assign a short string.

Reply:
So the stuff happening in the for loop is what is getting timed, so priming the string to a different size will only happen to the first iteration of the 10,000-1,000,000 iterations that will be performed. That said, timing the resize case is interesting, so I added another test to each of these that checks what the round-trip resize operation looks like.

Reply:
Oh right, good point.